classes/class-apm-mautic-api.php

```diff
@@ -381,6 +387,66 @@ public static function mautic_get_contact_by_email( $email, $mautic_credentials
 		}
 
 		/**
+		 * Check if contact is exist in mautic
+		 *
+		 * @since 1.0.0
+		 * @param id $id contact ID.
+		 * @return void
+		 */
+		public static function is_contact_published( $id ) {
+
+			$mautic_credentials = AMPW_Mautic_Init::get_mautic_credentials();
+
+			if ( $mautic_credentials['expires_in'] < time() ) {
+				$grant_type = 'refresh_token';
+				$response = self::mautic_get_access_token( $grant_type );
+				if ( is_wp_error( $response ) ) {
+					$error_msg = $response->get_error_message();
+					$status = 'error';
+					echo __( 'There appears to be an error with the configuration.', 'automateplus-mautic-wp' );
+				} else {
+					$access_details = json_decode( $response['body'] );
+					$expiration = time() + $access_details->expires_in;
+					$mautic_credentials['access_token'] = $access_details->access_token;
+					$mautic_credentials['expires_in'] = $expiration;
+					$mautic_credentials['refresh_token'] = $access_details->refresh_token;
+					update_option( 'ampw_mautic_credentials', $mautic_credentials );
```

Comment:
Like the `AMPW_Mautic_Init::get_mautic_credentials()` there should be `AMPW_Mautic_Init::update_mautic_credentials()` in future.